src/Receiver.h

```diff
@@ -271,22 +271,33 @@ class Receiver {
         /**
          * MessageBucket constructor.
          *
-         * @param messageTimeoutCycles
-         *      Number of cycles of inactivity to wait before a Message is
-         *      considered failed.
+         * @param receiver
+         *      Receiver that owns this bucket.
          * @param resendIntervalCycles
          *      Number of cycles of inactivity to wait between requesting
          *      retransmission of un-received parts of a Message.
          *      liveness of a Message.
          */
-        explicit MessageBucket(uint64_t messageTimeoutCycles,
+        explicit MessageBucket(Receiver* receiver,
                                uint64_t resendIntervalCycles)
-            : mutex()
+            : receiver(receiver)
+            , mutex()
             , messages()
-            , messageTimeouts(messageTimeoutCycles)
             , resendTimeouts(resendIntervalCycles)
         {}
 
+        /**
+         * Destruct a MessageBucket. Will destroy all contained Message objects.
+         */
+        ~MessageBucket()
+        {
+            // Intrusive::List is not responsible for destructing its elements;
+            // it must be done manually.
+            for (auto& message : messages) {
+                receiver->messageAllocator.destroy(&message);
```

Comment:
Interface feels like it's leaking and not really symmetric; the `MessageBucket` isn't involved when a message is allocated but is involved when it's destructed?

Reply:
I have decided to move the real work into `~Receiver()`